crates/orderbook/src/orderbook.rs

```diff
@@ -403,6 +406,66 @@ impl Orderbook {
             .await
             .context("get_user_orders error")
     }
+
+    pub async fn get_order_status(&self, uid: &OrderUid) -> Result<Option<dto::order::Status>> {
+        match self.database.latest_order_event(uid).await? {
+            None => Ok(None),
+            Some(event) => {
+                let fetch_solutions = || async move {
+                    let competition = match self.database.load_latest_competition().await {
+                        Ok(competition) => competition.common,
+                        Err(LoadSolverCompetitionError::NotFound) => {
+                            tracing::warn!("no latest competition exists");
+                            return Ok(Vec::new());
+                        }
+                        Err(err) => {
+                            tracing::error!(?err, "could not load latest competition");
+                            return Ok(Vec::new());
+                        }
+                    };
+                    let solutions = competition
+                        .solutions
+                        .into_iter()
+                        .map(|solution| {
+                            let executed_amounts = solution.orders.iter().find_map(|o| match o {
+                                solver_competition::Order::Legacy { .. } => None,
+                                solver_competition::Order::Colocated {
+                                    id,
+                                    sell_amount,
+                                    buy_amount,
+                                } => (id.0 == uid.0).then_some(dto::order::ExecutedAmounts {
+                                    sell: *sell_amount,
+                                    buy: *buy_amount,
+                                }),
+                            });
+                            dto::order::SolutionInclusion {
+                                solver: solution.solver,
+                                executed_amounts,
+                            }
+                        })
+                        .collect::<Vec<_>>();
+
+                    Ok::<Vec<_>, anyhow::Error>(solutions)
+                };
+                let status = match event.label {
+                    OrderEventLabel::Ready => dto::order::Status::Active,
+                    OrderEventLabel::Created => dto::order::Status::Scheduled,
+                    OrderEventLabel::Considered => {
+                        dto::order::Status::Solved(fetch_solutions().await?)
```

Comment:
This will not return the latest competition based on the position where the `OrderEventLabel::Considered` is stored in the Runloop (before solver competition table is saved to database)?

Reply:
> Even though FE is lucky enough to fetch the status when the competition is not in the DB yet, a subsequent request should fetch the actual data.

Makes sense, although I'm not convinced backend should make any assumptions on how the FE would use it. Also, in the future, we might add some code between these two calls that could take more time and break this assumption.

I am not that worried that `Considered` would be missing data in this edge case, but rather that it could contain wrong data (from previous partial execution of this trade, or something similar), and then, some of the subsequent runloop code might fail and end the iteration early, leaving the order status on FE in inaccurate state.